### PR TITLE
Add note about reflinks in tutorial

### LINKED ISSUE #139: add a note about reflinks and how space is calculated in the tutorial
In our article https://dvc.org/doc/tutorial/define-ml-pipeline , we use `du` to show that space taken by the repo didn't double. That works with hardlinks, but `du` is not smart enough when it comes to reflinks, which it counts as separate full-blown filse and accounts for their size two times(for the cache file and for the reflink), which makes it confusing for the mac users https://github.com/iterative/dvc/issues/1442 . We should add a note about it.

### PR DESCRIPTION
**Pull request by kurianbenoy**
Fix #139 

If you want to change the message, please let me know. Also should we mention about the innacurate space calculation by du command?
I am GSOD applicant as well 😊

### CODE REVIEW

File: static/docs/tutorial/define-ml-pipeline.md [hunk 1]
```diff
@@ -118,6 +118,9 @@ and Windows. Some implementation details (like inodes) might differ, but the
 overall DVC behavior is the same. See also [DVC Files and
 Directories](/doc/user-guide/dvc-files-and-directories)
```
Comment (shcheklein) [inline static/docs/tutorial/define-ml-pipeline.md]: this note above should be update to mention reflinks on modern file systems. I'm not sure why do we need the reference to the DVC files and directories here.

File: static/docs/tutorial/define-ml-pipeline.md [hunk 2]
```diff
@@ -118,6 +118,9 @@ and Windows. Some implementation details (like inodes) might differ, but the
...
 
+> Note: to prove that no duplication occurs, we could use df utility to see that
```
Comment (shcheklein) [inline static/docs/tutorial/define-ml-pipeline.md]: we need to mention that we need to refer to df only in case of reflinks.

Comment (shcheklein) [reply static/docs/tutorial/define-ml-pipeline.md]: I would also put this note above the example, so that a user can see it in advance

**Comment by shcheklein** [comment]
> Fix #139
> 
> If you want to change the message, please let me know. Also should we mention about the innacurate space calculation by du command?
> I am GSOD applicant as well 😊

Yes, I would mention that du is inaccurate in case of a modern file system that is using reflinks.

File: static/docs/tutorial/define-ml-pipeline.md (hunk 1 above)
Comment (kurianbenoy) [reply static/docs/tutorial/define-ml-pipeline.md]: Can I remove the reference to DVC files and directories?

Comment (shcheklein) [reply static/docs/tutorial/define-ml-pipeline.md]: @kurianbenoy Yes! I think it made it here by mistake. This link is not relevant. There should be a link to some doc that explains everything about link types, something like that one I shared below.

File: static/docs/tutorial/define-ml-pipeline.md [hunk 3]
```diff
@@ -15,6 +15,11 @@ to download `data.xml`and save it into the `data` subdirectory.
 
 </details>
 
+> Note: In case of systems supporting reflink, use `df` utility to see that free 
+space on the drive didn't decline by the file size that we are adding and no 
+duplication takes place. As `du` is inaccurate in modern file systems using
+reflinks.
```
Comment (shcheklein) [inline static/docs/tutorial/define-ml-pipeline.md]: Looks great! But it should go right before the code block that includes `du -sh .`. The point is to show that even we now have a file in the working space *and* in the `.dvc/cache` they still take the same amount of space.

Comment (kurianbenoy) [reply static/docs/tutorial/define-ml-pipeline.md]: Thanks, incorporated the feedback.

**Comment by kurianbenoy** [comment]
> Just move the paragraph a bit. Otherwise looks great! Thank you.

Made the changes, @shcheklein I highly value the quick response in dvc.org :smile:

**Pull request merged**